server/src/main/java/org/apache/iotdb/db/query/control/QueryTimeManager.java

```diff
@@ -90,20 +90,18 @@ public AtomicBoolean unRegisterQuery(long queryId) {
         queryId,
         (k, v) -> {
           successRemoved.set(true);
-          return null;
-        });
-    queryScheduledTaskMap.computeIfPresent(
-        queryId,
-        (k, v) -> {
-          queryScheduledTaskMap.get(queryId).cancel(false);
+          ScheduledFuture<?> scheduledFuture = queryScheduledTaskMap.remove(queryId);
+          if (scheduledFuture != null) {
+            scheduledFuture.cancel(false);
+          }
           return null;
         });
     return successRemoved;
   }
 
   public static void checkQueryAlive(long queryId) {
-    if (getInstance().queryInfoMap.get(queryId) != null
-        && getInstance().queryInfoMap.get(queryId).isInterrupted()) {
+    QueryInfo queryInfo = getInstance().queryInfoMap.get(queryId);
+    if (queryInfo != null && queryInfo.isInterrupted()) {
       if (getInstance().unRegisterQuery(queryId).get()) {
```

Comment:
hi,   getInstance().unRegisterQuery(queryId) maybe return null, so #get() will throw nullpointerexception

Reply:
It may return false but never return null. Am i wrong?

Reply:
It may return false but never return null. Am i wrong?

Reply:
sorry, I see fault..   map k-v return null...  
unRegisterQuery return boolean..
you are right...